components/script/dom/customelementregistry.rs

```diff
@@ -95,7 +96,15 @@ impl CustomElementRegistryMethods for CustomElementRegistry {
         rooted!(in(global_scope.get_cx()) let constructor = constructor_.callback());
 
         // Step 1
-        if unsafe { !IsConstructor(constructor.get()) } {
+        // We must unwrap the constructor as all wrappers are constructable if they are callable.
+        rooted!(in(global_scope.get_cx()) let unwrapped_constructor = unsafe { UnwrapObject(constructor.get(), 1) });
```

Comment:
Ok, in that case I think this should be ok without storing the unwrapped constructor. If we did go the Gecko route of using two different HashMaps with different keys, that locks us into using `Rc` even more. If we do go that route, that could be saved for a followup.